wakuv2/nwaku_test.go

```diff
@@ -161,80 +154,87 @@ func parseNodes(rec []string) []*enode.Node {
 // Using Docker:
 //
 //	IP_ADDRESS=$(hostname -I | awk '{print $1}');
-//	docker run \
-//	 -p 60000:60000/tcp -p 9000:9000/udp -p 8645:8645/tcp harbor.status.im/wakuorg/nwaku:v0.31.0 \
-//	 --tcp-port=60000 --discv5-discovery=true --cluster-id=16 --pubsub-topic=/waku/2/rs/16/32 --pubsub-topic=/waku/2/rs/16/64 \
-//	 --nat=extip:${IP_ADDRESS} --discv5-discovery --discv5-udp-port=9000 --rest-address=0.0.0.0 --store
+// 	docker run \
+// 	-p 61000:61000/tcp -p 8000:8000/udp -p 8646:8646/tcp harbor.status.im/wakuorg/nwaku:v0.33.0 \
+// 	--discv5-discovery=true --cluster-id=16 --log-level=DEBUG \
+// 	--nat=extip:${IP_ADDRESS} --discv5-udp-port=8000 --rest-address=0.0.0.0 --store --rest-port=8646 \
 
 func TestBasicWakuV2(t *testing.T) {
-	nwakuInfo, err := GetNwakuInfo(nil, nil)
-	require.NoError(t, err)
-
-	// Creating a fake DNS Discovery ENRTree
-	tree, url := makeTestTree("n", parseNodes([]string{nwakuInfo.EnrUri}), nil)
-	enrTreeAddress := url
-	envEnrTreeAddress := os.Getenv("ENRTREE_ADDRESS")
-	if envEnrTreeAddress != "" {
-		enrTreeAddress = envEnrTreeAddress
+	extNodeRestPort := 8646
+	storeNodeInfo, err := GetNwakuInfo(nil, &extNodeRestPort)
+	require.NoError(t, err)
+	
+	nwakuConfig := WakuConfig{
+		Port:        30303,
+		NodeKey:     "11d0dcea28e86f81937a3bd1163473c7fbc0a0db54fd72914849bc47bdf78710",
+		EnableRelay: true,
+		LogLevel:    "DEBUG",
+		DnsDiscoveryUrl: "enrtree://AMOJVZX4V6EXP7NTJPMAYJYST2QP6AJXYW76IU6VGJS7UVSNDYZG4@boot.prod.status.nodes.status.im",
+		DnsDiscovery: true,
+		Discv5Discovery: true,
+		Staticnodes: []string{storeNodeInfo.ListenAddresses[0]},
+		ClusterID: 16,
+		Shards: []uint16{64},
 	}
-
-	config := &Config{}
-	setDefaultConfig(config, false)
-	config.Port = 0
-	config.Resolver = mapResolver(tree.ToTXT("n"))
-	config.DiscV5BootstrapNodes = []string{enrTreeAddress}
-	config.DiscoveryLimit = 20
-	config.WakuNodes = []string{enrTreeAddress}
-	w, err := New(nil, "", config, nil, nil, nil, nil, nil)
+	
+	w, err := New(nil, "", &nwakuConfig, nil, nil, nil, nil, nil)
 	require.NoError(t, err)
 	require.NoError(t, w.Start())
 
 	enr, err := w.ENR()
 	require.NoError(t, err)
 	require.NotNil(t, enr)
 
-	// DNSDiscovery
-	ctx, cancel := context.WithTimeout(context.TODO(), 30*time.Second)
-	defer cancel()
-
-	discoveredNodes, err := dnsdisc.RetrieveNodes(ctx, enrTreeAddress, dnsdisc.WithResolver(config.Resolver))
-	require.NoError(t, err)
-
-	// Peer used for retrieving history
-	r, err := rand.Int(rand.Reader, big.NewInt(int64(len(discoveredNodes))))
-	require.NoError(t, err)
-
-	storeNode := discoveredNodes[int(r.Int64())]
-
 	options := func(b *backoff.ExponentialBackOff) {
 		b.MaxElapsedTime = 30 * time.Second
 	}
 
 	// Sanity check, not great, but it's probably helpful
 	err = tt.RetryWithBackOff(func() error {
-		if len(w.Peers()) < 1 {
-			return errors.New("no peers discovered")
+		
+		numConnected, err := w.GetNumConnectedPeers()
+		fmt.Println("numConnected: ", numConnected)
```

Comment:
Might be better to use a log here instead.

Reply:
Oops forgot about that! I added it for debugging purposes but it actually can spam a lot, so if it's ok I think I'll just get rid it.

Thank you!